CIP6/CIP6.md

```diff
@@ -0,0 +1,885 @@
+---
+CIP: 6
+Title: Extended Metadata
+Authors: Mike Fullman <mike@fullman.net>, Markus Gufler <gufmar@gmail.com>
+Comments-URI:
+Status: Draft
+Type: Standards
+Created: 2020-07-20
+License: Apache-2.0
+---
+
+## Abstract
+
+This CIP defines the concept of extended metadata for pools that is referenced from the pool registration data stored on chain.
+
+## Motivation
+
+As the ecosystem around Cardano stake pools proliferate so will the desire to slice, organize and search pool information dynamically.  Currently the metadata referenced on chain provides 512 bytes that can be allocated across the four information categories ([delegation-design-specification Section 4.2)](https://hydra.iohk.io/build/790053/download/1/delegation_design_spec.pdf):
```

Comment:
Note that the `512` bytes limitation only exists in softwares currently dealing with metadata (i.e. cardano-wallet and SMASH). Having a max size that is rather constrained prevents some DOS attacks on softwares consuming metadata. Yet, it should be possible to increase that limit to makes it possible to include more information, while still keeping a reasonable size.

Reply:
> Note that the `512` bytes limitation only exists ...  it should be possible to increase that limit to makes it possible to include more information, while still keeping a reasonable size.

it is an interesting option indeed, that should be considered as a possible way to go. In that case it any change requires a re-registration on chain with the new hash.  Having it only linked from main to the extended json file makes it more flexible but also a little bit less trusted.

Reply:
The limit for on chain data is great for those of us parsing, doubling it would be fine but I do think keeping it constrained and allowing the extended metadata to be the place for flexible and fast changes is the right tradeoff.

Reply:
> The limit for on chain data is great for those of us parsing, doubling it would be fine but I do think keeping it constrained and allowing the extended metadata to be the place for flexible and fast changes is the right tradeoff.

Agreed, it's way better to have a strictly defined set of _mandatory_ metadata and more flexible set of _extended_ metadata, while _mandatory_ metadata may be subject to extensive CIP review process, _extended_ metadata provide the ability to innovate with after-the-fact formalization thru the CIP process for greater interoperability.